src/ecmascript_simd.js

```diff
@@ -474,13 +474,13 @@ if (typeof SIMD.Bool32x4.select === "undefined") {
   }
 }
 
-if (!Object.hasOwnProperty(SIMD.Bool32x4.prototype, 'valueOf')) {
+if (!SIMD.Bool32x4.prototype.hasOwnProperty('valueOf')) {
```

Comment:
I definitely think that sort of change should happen, but not as part of this PR - this is fine to go in as-is.